### DISCUSSION TITLE
Is there a built in way to setup a global programming language switcher?

### DISCUSSION THREAD

**Discussion by janaka** [squidfunk/mkdocs-material#5297] [Q&A]
Is there a built-in way to setup a global programming language switcher?

We have created a docs site with a large number of MD files (1000+) with API refs and code examples. Currently in only one programming language. But want to add more, up to 4 more langs.

The docs are all generated. The current structure organises in folders named by language.

I think Material for MKdocs supports tabbed code blocks but these docs also have property and method names as inline code. So tabs start getting clunky.

So ideally there's a global switch. 

Any suggestion on how to easily implement this using existing plug-ins and features?

**Comment by squidfunk** [comment]
Besides [linked content tabs](https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#linked-content-tabs), there's currently no global switch, no. You can implement this yourself in JavaScript, but YMMV.

**Reply by janaka** [reply]
Thanks for the quick response. 

Any examples out there you can think of that might give me a jump-off point? something that folders navigating by folder structure maybe.

**Reply by squidfunk** [reply]
Not really, but any frontend-savvy developer should be able to pull something off. Material for MkDocs is hackable and extensible. I recommend reading our [customization guide](https://squidfunk.github.io/mkdocs-material/customization/) and searching the discussion board and issue tracker 😉

**Reply by janaka** [reply]
And ChatGPT 😊

**Comment by kamilkrzyskow** [comment]
If I understand you correctly, you want a global version switch that would change the current URL path to a different path based on the language, and the docs with different languages are in different directories. If you're not using `mike` for versioning you can "hack" a version switcher together using JavaScript extensions.

In our site we're using it to switch between the `main` and `dev` build that are hosted on different servers. It's basically the same principle, as path change.
https://github.com/Gothic-Modding-Community/gmc/blob/8f96e7fd2218554aabe7d0b9433b4e662993c7d8/overrides/assets/javascripts/extra.js#L264

**Reply by janaka** [reply]
Either mike or i18n were the other hacks in mind.
